Add topic properties reference (#1290)
redpanda-data/documentation: docs/manage/cluster-maintenance/disk-utilization.mdx

| Retention property | Cluster level | Topic level <br /> (overrides cluster configuration) |
| ---               | ---                                                |  ---                                 |
| Time-based        | [delete_retention_ms](../../../reference/cluster-properties#delete_retention_ms) | [retention.ms](../../../reference/topic-properties#retentionms)    |
| Size-based        | [retention_bytes](../../../reference/cluster-properties#retention_bytes) | [retention.bytes](../../../reference/topic-properties#retentionbytes) |

> | Segment size      | [log_segment_size](../../../reference/tunable-properties#log_segment_size) | [segment.bytes](../../../reference/topic-properties#segmentbytes)                                |

Reviewer: Hi @shorgi. Could you please pick up the latest file for this? It now includes `retention.local.target.ms` and `retention.local.target.bytes`. I'd removed the links and kept defaults in there since we couldn't link to the topic properties yet, but with topic properties linkable now, we can remove the defaults from this table!

rpk topic alter-config <topic> --set segment.bytes=<segment_size>
```

### Segment size for compacted topics
 

> Compaction, or key-based retention, saves space by retaining at least the most recent value for a message key within a topic partition's log and discarding older values. Compaction runs periodically in the background in a best effort fashion, and it doesn't guarantee that there are no duplicate values per key.

Reviewer 2: ```suggestion
Compaction, or key-based retention, saves space by retaining at least the most recent value for a message key within a topic partition's log and discarding older values. Compaction runs periodically in the background in a best effort fashion, and doesn't guarantee that there are no duplicate values per key.
```

When compaction is configured, topics take their size from `compacted_log_segment_size`. The `log_segment_size` property does not apply to compacted topics. 

Setting a `segment.bytes` size on a topic applies whether the topic is compacted or not, and the `max_compacted_log_segment_size` property applies to compacted topics regardless of any other properties. The `max_compacted_log_segment_size` property controls how many segments are merged together. For example, if you set `segment.bytes` to 128 MB, but leave `max_compacted_log_segment_size` at 5 GB, then you get 128 MB segments when they're written, but up to 5 GB segments after compaction.

> Redpanda does compaction periodically in the background. The compaction period is configured by the cluster property [log_compaction_interval_ms](../../../reference/cluster-properties#log_compaction_interval_ms).

Reviewer 2: ```suggestion
Redpanda periodically performs compaction in the background. The compaction period is configured by the cluster property [log_compaction_interval_ms](../../../reference/cluster-properties#log_compaction_interval_ms).
```

> ### Segment size for compacted segments

Reviewer 3: Did you mean to write "compacted topics" instead of "compacted segments" here?

 
Compaction, or key-based retention, saves space by retaining only the last known value for each message key within the segment. The latest value for each key is preserved, and all older values are discarded. 

> 

Reviewer 4: Here we may mention that this is a best effort, and there still may be duplicates

When compaction is configured, topics take their size from `compacted_log_segment_size`. The `log_segment_size` property does not apply to compacted topics. 

Setting a `segment.bytes` size on a topic applies whether the topic is compacted or not, and the `max_compacted_log_segment_size` property applies to compacted topics regardless of any other properties. The `max_compacted_log_segment_size` property controls how many segments are merged together. For example, if you set `segment.bytes` to 128 MB, but leave `max_compacted_log_segment_size` at 5 GB, then you get 128 MB segments when they're written, but up to 5 GB segments after compaction.

Redpanda does compaction periodically in the background. The compaction period is configured by the cluster property [log_compaction_interval_ms](../../../reference/cluster-properties#log_compaction_interval_ms).

> Keep in mind that very large segments prevent compaction, and very small segments increase the risk of encountering resource limits. To calculate an upper limit on segment size, divide the disk size by the number of partitions. For example, if you have a 128 GB disk and 1000 partitions, the upper limit of the segment size is `134217728`. Default is `1073741824`. 

Reviewer 3: > Keep in mind that very large segments prevent compaction

I think this is important to consider, on a first glance it seemed that this section however does not explain why that is right? In case I have not just missed it I could imagine linking to the place where the compaction process is explained (segments which are eligible for compaction specifically).

Author: @weeco and @micheleRP (I think you updated this last?) - I think large segments don't always prevent compaction but slow it down. See suggestion:

```suggestion
Keep in mind that very large segments delay, or possibly prevent, compaction. A very large active segment cannot be cleaned up or compacted until it is closed, and very large closed segments require significant memory and CPU to process for compaction. Very small segments increase the frequency of processing for applying compaction and resource limits. To calculate an upper limit on segment size, divide the disk size by the number of partitions. For example, if you have a 128 GB disk and 1000 partitions, the upper limit of the segment size is `134217728`. Default is `1073741824`. 
```

Also, the docs are missing an explanation of the compaction concept (and other Kafka topic concepts). That'll be added in a separate PR and linked to here.

Reviewer 3: That's a good addition I think. After reading this the only question I'd still have - what's the problem with frequent processing of very small segments (and what does very small mean here - like one segment per minute or more segments)? Seems fine to me to have compact "frequently" or are we saying that this ist just way more inefficient rather than finding a reasonable sized segment?
